### PR TITLE
Added functionality to location button and the forms 

### PR DESCRIPTION
**Pull request by JChen0915**
The location button now takes the geolocation of the user and updates to firestore when Update Preferences is pressed. 

The two dropdown forms for location limit and yelp rating limit also add their information to firestore.

<Preferences.js>
Aforementioned changes of adding functionality

The preferencesMap global variable has been removed, as categorical preferences are now stored as an array of currently selected categories, making the map no longer necessary.

### CODE REVIEW

File: my-app/src/Components/Preferences.js
Comment (chizy7) [inline my-app/src/Components/Preferences.js:1]: Overall, the changes made in Preferences.js look good and accomplish the intended tasks.

Consider using a more meaningful name for the state variable `value`. A name like `activeCategories` would better represent its purpose.

**Review by chizy7 — APPROVED** [review]
Great job!

**Review by Kirazuto7 — APPROVED** [review]
Changes look good. I recommend maybe picking some better variable name(s).

**Review by caid4134 — APPROVED** [review]
Vastly improved functionality of the preference page but as others have said, future maintenance may be a bit more difficult for others.

**Pull request merged**